core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -181,6 +185,34 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
       else consumer.pause(java.util.Collections.singleton(tp))
     }
 
+    def tryPoll(timeout: Long, onSuccess: ConsumerRecords[K, V] => Unit): Unit = {
```

Comment:
what about returning `Try[ConsumerRecords]` here instead of passing `onSuccess`?
This also allow us to drop `recover` and use just one `transform`